File: bench/alloc_bench/alloc_bench.cpp

#include <stdlib.h>
#include <vector>

int* pint(int x) {
  int* i = (int*)malloc(sizeof(int));
  *i = x;
  return i;
}

int main() {
  for (int i = 0; i < 1000000; i++) {
    std::vector<int*> v;
    v.push_back(pint(1));
    v.push_back(pint(2));
    v.push_back(pint(3));
    v.push_back(pint(4));
    v.push_back(pint(5));
  }
}

File: bench/alloc_bench/alloc_bench.go

package main

import "runtime"

func pint(x int) *int {
  i := new(int)
  *i = x
  return i;
}

func main() {
  for i := 0; i < 1000000; i++ {
    v := make([]*int, 0, 8)
    v = append(v, pint(1))
    v = append(v, pint(2))
    v = append(v, pint(3))
    v = append(v, pint(4))
    v = append(v, pint(5))
    _ = v
  }
  runtime.GC()
}
